apps/ui-patterns/src/app/UIPatternsWizard/steps/LastStep.tsx

```diff
@@ -12,16 +15,37 @@ export const LastStep: React.FC = () => {
     error: null,
   });
 
+  const { data: formData } = useWizardValues();
+
   const onSubmit = useCallback(() => {
     setSubmitResult({ data: 'Submitted!', isLoading: false, error: null });
   }, [setSubmitResult]);
+
+  const valid = useMemo(() => {
+    try {
+      const concatSchema = [
+        stepOneValidationSchema,
+        stepTwoValidationSchema,
+        stepThreeValidationSchema,
+      ].reduce((previous, current) => previous.concat(current), Yup.object());
```

Comment:
I have never used array reductions. So I assume this is doing something smarter than I understand.

Does that code do something that
```
[].concat( stepOneValidationSchema, stepTwoValidationSchema, stepThreeValidationSchema )
```
does not?

Reply:
this is a reducer on the elements of the array (where the reduction function is Yup.object.concat)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce

Reply:
Thanks. I didn't recognize the `previous.concat` was `Yup.object.concat`, not `Array.prototype.concat`.